src/sentry/testutils/skips.py

```diff
@@ -55,6 +57,13 @@ def _service_available(host: str, port: int) -> bool:
         return True
 
 
+def _service_options(name: str) -> dict[str, Any]:
+    options = settings.SENTRY_DEVSERVICES[name](settings, {})
+    options["name"] = f"sentry_{name}"
```

Comment:
this isn't safe since you aren't copying - you're mutating part of settings.SENTRY_DEVSERVICES and you'll get things like sentry_name_name_name as this is called more than once

Reply:
My thinking is: `SENTRY_DEVSERVICES` is dictionary of service name to functions.

```
SENTRY_DEVSERVICES: dict[str, Callable[[Any, Any], dict[str, Any]]] = {
    "redis": lambda settings, options: (
        {
            "image": ".....",
             .....
        }
    )
```

That means this code is getting the `SENTRY_DEVSERVICES` dictionary, looking up the function for a service and then calling that function to get a new direction containing the service config.

The only reason I set the name, is because this is something devservices does in [_prepare_containers](https://github.com/getsentry/sentry/blob/1652d3cd006952326cef800f63d58ea0e73ceeef/src/sentry/runner/commands/devservices.py#L328) which works in a similar fashion to this.

Am I misunderstanding this and actually the lambda function is return a single instance of the config?

Reply:
oh forgive me, you're right